File: 2022/sample5.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

import cv2

# (1)指定された番号のカメラに対するキャプチャ構造体を作成する
capture = cv2.VideoCapture(0)

# (2)表示用ウィンドウをの初期化
cv2.namedWindow("Capture")

while True:
    # (3)カメラから画像をキャプチャする
    ret, frame = capture.read();

    # (4) カメラ画像の表示
    cv2.imshow("Capture", frame)

    c = cv2.waitKey(2)
    if c == 27: # Escキー
        break

capture.release()
cv2.destroyAllWindows()

File: 2022/sample5.cpp
#include <opencv2/core/core.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/highgui/highgui.hpp>

int main (int argc, char **argv)
{
    cv::Mat frame;
    unsigned char c;

    // (1)指定された番号のカメラに対するキャプチャオブジェクトを作成する
    cv::VideoCapture capture(0);

    // (2)表示用ウィンドウをの初期化
    cv::namedWindow ("Capture", cv::WINDOW_AUTOSIZE);

    while (capture.isOpened()) {

      // (3)カメラから画像をキャプチャする
      capture.read(frame);

      // (4) カメラ画像の表示
      cv::imshow ("Capture", frame);

      // (5) 2msecだけキー入力を待つ
      c = cv::waitKey (2);
      if (c == '\x1b') // Escキー
        break;
    }

    cv::destroyWindow ("Capture");

    return 0;
}
